### ISSUE TITLE
Add image resource as an input resource

### ISSUE THREAD

**Issue by nader-ziada** [tektoncd/pipeline#260]
## Expected Behavior
When I define an input resource of type image to my task, the taskRun (build) created should have this image as an input source and fetch it

## Actual Behavior
Currently the input source of image is not added to the build input source

## Additional Info
Should come after #200 is done

**Comment by bobcatfish** [comment]
@pivotal-nader-ziada if you define an image as an input, would the contents of the image end up mounted somewhere? or would it actually be expected to be started and running? or something else?

**Comment by pivotal-nader-ziada** [comment]
I was expecting it to be available in the container for the task steps to use somehow

**Comment by bobcatfish** [comment]
> I was expecting it to be available in the container for the task steps to use somehow

I am tempted to hold off on doing this until we have some clear use cases where we want images to work as an input :thinking: 

Or we could go ahead and assume that this means we want the content of the image mounted into `workspace`?

(At the moment, esp. with how #224 is going, it will be almost impossible to run an image built in a previous step :S)

**Comment by bobcatfish** [comment]
> (At the moment, esp. with how #224 is going, it will be almost impossible to run an image built in a previous step :S)

Scratch that, I got confused, #224 is about steps within a Task, we could probably still have a future Task use an image built by a previous Task

**Comment by pivotal-nader-ziada** [comment]
I agree this can wait for a future milestone

**Comment by afrittoli** [comment]
We may want to revisit this in light of the current work on resources https://github.com/tektoncd/pipeline/issues/1076.

I think we need to decide whether a Resource is something that maps a state from outside to into a task from and from a task back to outside - or if we want resources to do actual work, like building an image or deploy an application.

It would be interesting to have an abstraction for common tasks like building an image or deploying an application, but perhaps those could simply be Tasks from the catalog, and we only need a way to refer to a Task from the catalog.

**Comment by afrittoli** [comment]
One use case for resources I've been thinking about it some form of input image resource: it's quite common to want a custom image for a step and I think we need to have a better solution for that. One idea could be having the step image specified as an input - i.e. a pipeline builds the step image(s) and produces output image resources which include the digest. Then other pipelines that consume those images take the resources as inputs

**Comment by bobcatfish** [comment]
Closing this in favor of https://github.com/tektoncd/pipeline/issues/1673 since we're redesigning all the PipelineResources (including this one) as part of that.
